### DISCUSSION TITLE
Put `pull_request` details in the event payload for `merge_group`

### DISCUSSION THREAD

**Discussion by yogurtearl** [community/community#65239] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

For workflows triggered by `merge_group`, it would be really helpful to have the information about the pull request(s) in the group via the `github.event` context, AND in the webhook payload. 

https://docs.github.com/en/webhooks/webhook-events-and-payloads#merge_group

A `merge_group` event is in response to one (or more) pull requests, for a specific event it would be useful to have the metadata for the pull request(s) that are being merged in the current group. 

related discussions:
- https://github.com/orgs/community/discussions/62219
- https://github.com/orgs/community/discussions/41939
- https://github.com/orgs/community/discussions/41939

**Comment by queenofcorgis** [comment]
Welcome to the GitHub Community, @yogurtearl , we're happy you're here! To ensure this post gets to the proper team, I've moved this to our Pull Requests category.

**Comment by neil176** [comment]
My team is also interested in this. A significant benefit of including specifically a PR number in the event context would be that we could use it in the concurrency configuration of our merge group workflow. 

Additionally we have had to add custom instrumentation to tie merge group workflows back to PRs, and it seems like that shouldn't be necessary since the PR number is already known on the github side as shown in the merge group branch naming.

**Comment by jasondamour** [comment]
@queenofcorgis Checking in, whats the status of this? **Is there any solution for merge queue concurrency today?** Trying to set a concurrency key which takes the `merge_group` into account isn't possible today. There is a `merge_group` "`destroyed`" event, but there isn't any way to consume it today afaik. So whats the solution?

**Comment by joehorsnell** [comment]
Hi @queenofcorgis (or any other GitHub employees) - can you provide any update on this please?

This would be a genuinely useful addition and quality-of-life improvement for teams adopting merge queues.

In the [docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-github-actions) right now, it mentions that:

>you need to update the workflows to include the `merge_group` event as an additional trigger.

Then it shows the basic workflow triggering stanza:

```
on:
  pull_request:
  merge_group:
```

However, it doesn't mention at all that the `merge_group` event payload is very sparse compared to the `pull_request` one, so if your workflows make use of the pull request info, or any other data missing from the `merge_group` payload, you are out of luck and have to DIY.

The [pr-details-action](https://github.com/matrix-org/pr-details-action) can help, which has [support for merge queues](https://github.com/matrix-org/pr-details-action/commit/15bde5285d7850ba276cc3bd8a03733e3f24622a), but it's not clear why this is not included by GitHub in the payload already?

What's the rationale for _not_ having it? It's known what the PR is, as it's in the temp branch name - `gh-readonly-queue/<base>/pr-123-<sha>`
